### ISSUE TITLE
NULLS FIRST not implemented

### ISSUE THREAD

**Issue by prescience-data** [oguimbal/pg-mem#133]
Just noticed that `ORDER BY col NULLS FIRST` is not yet implemented.

Understandable if not on the agenda, but thought I'd flag it just in case.

Thanks!

**Comment by oguimbal** [comment]
Thanks :)

I'll add that to my todolist, but I dont have time to work on it these days :(

**Comment by gthb** [comment]
Looks like you ended up having time :) — this was implemented in 71d3471c87312d7e5d03e3025aaa24c55c06afbd.

**Comment by gthb** [comment]
Oh, but the defaulting is wrong. Per the docs:

> By default, null values sort as if larger than any non-null value; that is, `NULLS FIRST` is the default for `DESC` order, and `NULLS LAST` otherwise.

and psql:
```sql
select t.val as value from test t order by t.val asc nulls first;
```
yielding:
```
 value 
-------
 
 a
 b
```

so this test should pass:

```ts
    it('can order by asc with nulls first', () => {
        expect(many(`create table test(val text);
            insert into test values ('b'), ('a'), (null);
            select t.val as value from test t order by t.val asc nulls first`))
            .to.deep.equal([
                { value: null }
                , { value: 'a' }
                , { value: 'b' }
            ]);
    });
```

but it fails.

**Comment by gthb** [comment]
Posted a fix in https://github.com/oguimbal/pg-mem/pull/296/

**Comment by oguimbal** [comment]
should be fixed with  pg-mem@2.6.5  thanks
